backend/trafficrules/src/test/java/com/coffeejug/trafficrules/service/StatisticServiceTest.java

```diff
@@ -0,0 +1,31 @@
+package com.coffeejug.trafficrules.service;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+@ExtendWith(MockitoExtension.class)
+class StatisticServiceTest {
+
+    @Mock
+    private StatisticService statisticService;
+
+    @Test
+    void shouldReturnZero() {
```

Comment:
Since you have mocked `statisticService` this test doesn't test anything. It will just always return 0 as an uptime. And also, why should it return zero if you are waiting for 1 second in the test?

Reply:
IDK why, but it works.

Reply:
The fact that the test is green does not mean that it works. In this case, it doesn't test anything. Try to break the service code, for example:
```
public long getUptime() {
   throw new RuntimeException();
}
```
The test should fail in this case, but it will pass

Reply:
Yes, really, in this time getUptime() returns 0L. Can u explain why ?

Reply:
Because you are using a mock of `statisticService`. It's not the real implementation. By default the mock objects in mockito have the default return values: null for objects and the default value of primitives.
In a nutshell, mocks should use the dependencies of the object under test while the object itself should be the real implementation.

Reply:
I see. "public Long getUptime()" returns 0L also btw.

Reply:
Maybe. But it's not the issue here, this class should not be mocked at all.
One method to check that the test is correct is to break the production code (code in service) and the test should become red.

Reply:
@sergiykatriuk what about this fix?

Reply:
Haven't reached this yet.